test/libsolidity/lsp/references/import_aliases.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.0;
+
+import "../goto/lib.sol" as Imported;
+//                          ^ @ImportedDef
+
+contract C
+{
+    function one(uint lhs, uint rhs) public pure returns (uint result)
+    {
+        result = Imported.Lib.add(lhs, rhs + 123);
+        //       ^ @ImportedUse
+        //       ^^^^^^^^ @Imported1
+        //       ^^^^^^^^^^^^ @ImportedLib1
+    }
+    function two(uint lhs, uint rhs) public pure returns (uint result)
+    {
+        result = one(Imported.Lib.add(lhs, rhs), rhs);
+        //                    ^ @LibRef
+        //           ^^^^^^^^ @Imported2
+        //           ^^^^^^^^^^^^ @ImportedLib2
+    }
+}
+// ----
+// lib: @diagnostics 2072
+// -> textDocument/documentHighlight {
+//     "position": @ImportedDef
+// }
+// <- [
+//     { "kind": 2, "range": @Imported1 },
+//     { "kind": 2, "range": @Imported2 }
+// ]
+// -> textDocument/documentHighlight {
+//     "position": @ImportedUse
+// }
+// <- [
+//     { "kind": 2, "range": @Imported1 },
+//     { "kind": 2, "range": @Imported2 }
+// ]
+// -> textDocument/documentHighlight {
+//     "position": @LibRef
+// }
+// <- [
+//     { "kind": 2, "range": @ImportedLib1 },
+//     { "kind": 2, "range": @ImportedLib2 }
+// ]
```

Comment:
It didn't find the definition of `Lib` in the imported file. Is this intentional? I thought IDEs usually find references all over the project, not just in a single file?

For the highlight request the spec says "scoped to this file":
> For programming languages this usually highlights all references to the symbol scoped to this file. 

But on the other hand for find references request it says "project-wide references":
> The references request is sent from the client to the server to resolve project-wide references for the symbol denoted by the given text document position.

So should it be scoped to the file or project-wide?